### ISSUE TITLE
Desires around Error Handling

### ISSUE THREAD

**Issue by strega-nil** [nushell/nushell#3943]
### Related problem

I am trying to handle errors, and export errors, but don't have a good way to do so.

### Describe the solution you'd like

I want two (interrelated but distinct) things:

1. The ability to stream to and capture from the `stderr` stream. This, importantly, does not imply an error; writing to `stderr` for logging is common enough that it should not be treated like an error to have written to `stderr`. Other options exist for nu-shell programs, (see PowerShell's many streams), but I don't believe they add enough benefit for the complexity they add when dealing with interoperability with native programs. Stuff like `Write-Verbose` did not need to be a stream on its own, and neither did `Write-Warning`, imo. Therefore, I would like to be able to just write to this stream (perhaps via `echo --stderr`?)
2. The ability to set an error, and the ability to capture errors from both native and nu-shell programs. Ability to "throw" more complex errors is Fine, but not necessary (and it adds a lot of complexity). When a sub-program fails, if I did not explicitly capture errors in the calling program, I should also fail (and this should go up until one hits either the interactive shell, or errors are captured.

### Describe alternatives you've considered

see above.

### Additional context and details

Not sure. I'm sure the powershell method of handling errors would be useful to look at, although very much not to copy.

**Comment by fdncred** [comment]
an `error` command would be cool!

**Comment by hustcer** [comment]
+1

**Comment by fdncred** [comment]
btw - we can create an error in script now with this PR https://github.com/nushell/nushell/pull/3986

maybe we should close this then?

**Comment by hustcer** [comment]
> btw - we can create an error in script now with this PR #3986
> 
> maybe we should close this then?

Nope,  PR #3986 only make it possible to Create an error, but can not handle errors

**Comment by rgwood** [comment]
FYI v0.72 has some improvements in this area: https://www.nushell.sh/blog/2022-11-29-nushell-0.72.html#redirection

stderr can now be redirected like `cat missingfilename.txt err> my_error.txt`

There are also some error handling improvements with try/catch but I think there's still some work to be done on errors from external commands.
